### PR TITLE
feat: add cancel button to input type search

### PR DESCRIPTION
**Pull request by artmsilva**
## What I did
[SRED-145](https://arcpublishing.atlassian.net/jira/software/c/projects/SRED/boards/1185?modal=detail&selectedIssue=SRED-145)

[SRED-145]: https://arcpublishing.atlassian.net/browse/SRED-145?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ui/input-text/src/InputText.tsx [hunk 1]
```diff
@@ -186,24 +192,47 @@ export const InputText = React.forwardRef<HTMLInputElement, InputTextProps>(
...
+    let child: React.ReactNode;
     switch (type) {
       case "search":
-        child = (
-          <Icon label="">
-            <Search />
-          </Icon>
-        );
+        if (
+          // isFloating &&
```
Comment (artmsilva) [inline ui/input-text/src/InputText.tsx:218]: i should refactor this to use isFloating only

File: ui/input-text/src/InputText.tsx [hunk 2]
```diff
@@ -186,24 +192,47 @@ export const InputText = React.forwardRef<HTMLInputElement, InputTextProps>(
...
         onBlur,
-        onChange
+        onChange,
+        isAutofilled
       );
 
     function handleButtonIconClick(event) {
       onButtonIconClick && onButtonIconClick(event);
+
+      if (type === "search" && internalRef && internalRef.current) {
```
Comment (artmsilva) [inline ui/input-text/src/InputText.tsx:208]: refactor to use isFloating

**Review by hs4man21 — COMMENTED** [review]
Is the cancel button a native part of input field? I was unable to find it in the dom via dev tools.

File: ui/input-shared/src/InputShared.tsx [hunk 3]
```diff
@@ -109,13 +115,16 @@ export const useFloating = (
...
 
   React.useEffect(() => {
-    if (val && !isFloating) {
+    if (val || isAutofilled) {
       setIsFloating(true);
       setIsTouched(true);
-    } else if (!val && prevValue.current && !isFocused && isFloating) {
-      setIsFloating(false);
-      setIsTouched(false);
+    } else {
```
Comment (wp-aberg) [inline ui/input-shared/src/InputShared.tsx:121]: 🧼

**Review by ebgranger — APPROVED** [review]

**Review by wp-aberg — COMMENTED** [review]
There is one edge case I just found, if placeholder text is used the close x is visible

image

Not sure how important / real this case is but just in case

**Pull request merged**